app/models/tag.rb

```diff
@@ -54,8 +53,17 @@ def taggings_count_cache_key
   end
 
   def write_taggings_to_redis(value)
+    # Atomically set the value while extracting the old value.
+    old_redis_value = REDIS_GENERAL.getset("tag_update_#{id}_value", value)
+
+    # If the value hasn't changed from the saved version or the REDIS version,
+    # there's no need to write an update to the database, so let's just bail
+    # out.
+    return value if value == old_redis_value && value == taggings_count_cache
+
+    # If we've reached here, then the value has changed, and we need to make
+    # sure that the new value is written to the database.
     REDIS_GENERAL.sadd("tag_update", id)
```

Comment:
I think the only time it'd *really* matter is one of those 0/1 situations again. (e.g. Someone uses a tag for the very first time, and then the `taggings_count_cache` is set to 1. Then they accidentally delete the tag from their work, causing the `taggings_count` code to try to set it to 0. If they add back the tag quickly enough, the `taggings_count_cache` might stay at 0 and not show up in the wrangling interface.)

It's an extremely unlikely occurrence, but even so I figure it's good to prevent these sorts of issues from popping up. :)